src/storage/src/hummock/sstable/mod.rs

```diff
@@ -379,6 +379,42 @@ impl SstableMeta {
     }
 }
 
+pub struct SstableBlockIterator<'a> {
+    pub sstable: &'a Sstable,
+    pub current_block_id: usize,
+}
+
+impl<'a> SstableBlockIterator<'a> {
+    pub fn new(sstable: &'a Sstable) -> Self {
+        Self {
+            sstable,
+            current_block_id: 0,
+        }
+    }
+
+    #[inline(always)]
+    pub fn next(&mut self) {
+        self.current_block_id += 1;
+    }
+
+    #[inline(always)]
+    pub fn valid(&self) -> bool {
+        self.current_block_id < self.sstable.block_count()
+    }
+
+    pub fn current_block_smallest(&self) -> &[u8] {
```

Comment:
this function contradicts with `valid()`. Please remind the caller to call `valid()` before call this function